kotlinpoet/src/nonJvmMain/kotlin/com/squareup/kotlinpoet/Util.nonJvm.kt

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (C) 2024 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.kotlinpoet
+
+internal actual fun <K, V> Map<K, V>.toImmutableMap(): Map<K, V> =
+  toMap()
+
+internal actual fun <T> Collection<T>.toImmutableList(): List<T> =
+  toList()
+
+internal actual fun <T> Collection<T>.toImmutableSet(): Set<T> =
+  toSet()
+
+internal actual fun formatIsoControlCode(code: Int): String {
+  return buildString(6) {
+    append("\\u")
+    appendFormat04x(code)
+  }
+}
+
+@OptIn(ExperimentalStdlibApi::class)
+private val HexFormatWithoutLeadingZeros = HexFormat {
+  number {
+    removeLeadingZeros = true
+  }
+}
+
+@OptIn(ExperimentalStdlibApi::class)
+internal fun Appendable.appendFormat04x(code: Int) {
+  val hex = code.toHexString(HexFormatWithoutLeadingZeros)
+  if (hex.length < 4) {
+    repeat(4 - hex.length) { append('0') }
+  }
+  append(hex)
+}
+
+@OptIn(ExperimentalStdlibApi::class)
+internal actual fun Int.toHexStr(): String =
+  toHexString(HexFormatWithoutLeadingZeros)
```

Comment:
Is this just `toString(16)`? If so, we can use that directly in common code (without relying on unstable APIs, as well).

Reply:
Yes, it seems so. I'll simplify it

Reply:
[0920d3e](https://github.com/square/kotlinpoet/pull/2006/commits/0920d3e8a10b16f7b71484c7c642a48019ce2a7d)